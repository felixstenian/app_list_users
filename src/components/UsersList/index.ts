export { default as CardUser } from './CardUser'
export type { CardUserProps } from './CardUser'
export { default } from './UsersList'
export type { UsersListProps } from './UsersList'
