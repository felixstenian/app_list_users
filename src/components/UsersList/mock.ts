export const listMock = [
  {
    gender: 'female',
    name: {
      title: 'Miss',
      first: 'Jennie',
      last: 'Nichols'
    },
    location: {
      street: {
        number: 8929,
        name: 'Valwood Pkwy'
      },
      city: 'Billings',
      state: 'Michigan',
      country: 'United States',
      postcode: '63104',
      coordinates: {
        latitude: '-69.8246',
        longitude: '134.8719'
      },
      timezone: {
        offset: '+9:30',
        description: 'Adelaide, Darwin'
      }
    },
    email: 'jennie.nichols@example.com',
    login: {
      uuid: '7a0eed16-9430-4d68-901f-c0d4c1c3bf00',
      username: 'yellowpeacock117',
      password: 'addison',
      salt: 'sld1yGtd',
      md5: 'ab54ac4c0be9480ae8fa5e9e2a5196a3',
      sha1: 'edcf2ce613cbdea349133c52dc2f3b83168dc51b',
      sha256: '48df5229235ada28389b91e60a935e4f9b73eb4bdb855ef9258a1751f10bdc5d'
    },
    dob: {
      date: '1992-03-08T15:13:16.688Z',
      age: 30
    },
    registered: {
      date: '2007-07-09T05:51:59.390Z',
      age: 14
    },
    phone: '(272) 790-0888',
    cell: '(489) 330-2385',
    id: {
      name: 'SSN',
      value: '405-88-3636'
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/75.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/75.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/75.jpg'
    },
    nat: 'US'
  },
  {
    gender: 'female',
    name: {
      title: 'Mrs',
      first: 'Ruby',
      last: 'Jones'
    },
    location: {
      street: {
        number: 4803,
        name: 'Wiri Station Road'
      },
      city: 'Greymouth',
      state: 'Wellington',
      country: 'New Zealand',
      postcode: 17397,
      coordinates: {
        latitude: '54.2932',
        longitude: '157.3572'
      },
      timezone: {
        offset: '+5:30',
        description: 'Bombay, Calcutta, Madras, New Delhi'
      }
    },
    email: 'ruby.jones@example.com',
    login: {
      uuid: 'd55b7f3e-1330-41e5-9582-acd000035cf1',
      username: 'blackmouse796',
      password: 'kristina',
      salt: 'nmcf5uwv',
      md5: '2553b4d72b2ea7eee593b5791a6054bf',
      sha1: 'dc022ea1291986b281d7f21395ebffecf343db77',
      sha256: '7bc81c3c2198f5eb2c3030a6bf8d62cc8b4c432357e5e0f7c1b5f2395e5987a0'
    },
    dob: {
      date: '1993-08-10T17:00:07.860Z',
      age: 29
    },
    registered: {
      date: '2003-01-23T06:05:41.680Z',
      age: 20
    },
    phone: '(002)-379-0454',
    cell: '(311)-839-3611',
    id: {
      name: '',
      value: null
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/women/2.jpg',
      medium: 'https://randomuser.me/api/portraits/med/women/2.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/women/2.jpg'
    },
    nat: 'NZ'
  }
]
