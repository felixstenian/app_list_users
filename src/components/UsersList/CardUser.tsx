import { useState } from 'react'

import { Image, ModalUserDetails, Row, Text } from 'components'
import { User } from 'shared/types'

export type CardUserProps = {
  userData: User
}

const CardUser = ({ userData }: CardUserProps) => {
  const [openProfile, setOpenProfile] = useState<boolean>(false)

  const displayDataUser = () => setOpenProfile(true)
  const onCloseUserDetails = () => setOpenProfile(false)

  return (
    <>
      {openProfile && (
        <ModalUserDetails
          userData={userData}
          open={openProfile}
          onClose={onCloseUserDetails}
        />
      )}
      <Row
        key={userData?.login?.uuid}
        bg="secondary"
        width="80%"
        justifyContent="center"
        alignItems="center"
        p={10}
        borderRadius="8px"
        my="5px"
        gap="1rem"
        flexWrap="wrap"
        minWidth="32rem"
        as="a"
        onClick={displayDataUser}
        cursor="pointer"
      >
        <Row
          alignItems="center"
          justifyContent={['center', 'flex-start', 'flex-start']}
          flex={1}
        >
          <Image
            src={userData?.picture?.large}
            alt="alt"
            width={['10rem', '12rem', '8rem']}
            height={['10rem', '12rem', '8rem']}
            borderRadius="10rem"
          />
          <Text ml={[30, 20, 20]} variant="mediumTitle">
            {userData?.name?.first}
          </Text>
        </Row>
        <Row
          flexWrap="wrap"
          justifyContent="center"
          gap={['4rem', '3rem', '5rem']}
        >
          <Text variant="subtitle">{userData?.location?.country}</Text>
          <Text>{userData?.email}</Text>
        </Row>
      </Row>
    </>
  )
}

export default CardUser
