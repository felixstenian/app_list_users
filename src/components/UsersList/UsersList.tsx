import { Row } from 'components'

import { CardUser, User } from '.'

export type UsersListProps = {
  list: User[]
}

const UsersList = ({ list }: UsersListProps) => (
  <Row flexDirection="column" px={10}>
    {list.map((user) => (
      <CardUser key={user.id} user={user} />
    ))}
  </Row>
)

export default UsersList
