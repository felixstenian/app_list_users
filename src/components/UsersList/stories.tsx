import { Meta, Story } from '@storybook/react'
import UsersList, { CardUser, CardUserProps, UsersListProps } from '.'
import { list } from './mock'

export default {
  title: 'UsersList',
  component: UsersList,
  args: {
    list: list,
    user: {
      type: list[0]
    }
  }
} as Meta

export const Default: Story<UsersListProps> = (args) => <UsersList {...args} />

export const CardUserDefault: Story<CardUserProps> = (args) => (
  <CardUser {...args} />
)

CardUserDefault.args = {
  user: list[0]
}
