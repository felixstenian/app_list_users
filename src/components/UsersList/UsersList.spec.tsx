import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils'
import UsersList, { CardUser } from '.'
import { listMock } from './mock'

describe('<UsersList />', () => {
  it('1. should render the list of users', () => {
    const { container } = renderWithTheme(<UsersList listUsers={listMock} />)

    expect(screen.getByTestId('listUsers')).toBeInTheDocument()
    expect(screen.getByText('Jennie')).toBeInTheDocument()
    expect(screen.getByText('United States')).toBeInTheDocument()
    expect(screen.getByText('jennie.nichols@example.com')).toBeInTheDocument()
    expect(screen.getByText('Ruby')).toBeInTheDocument()
    expect(screen.getByText('New Zealand')).toBeInTheDocument()
    expect(screen.getByText('ruby.jones@example.com')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('2. should render the user information correctly', () => {
    renderWithTheme(<CardUser userData={listMock[0]} />)

    expect(screen.getByText('Jennie')).toBeInTheDocument()
    expect(screen.getByText('United States')).toBeInTheDocument()
    expect(screen.getByText('jennie.nichols@example.com')).toBeInTheDocument()
  })
})
