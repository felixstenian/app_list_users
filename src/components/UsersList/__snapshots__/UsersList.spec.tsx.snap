// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> 1. should render the list of users 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  cursor: pointer;
  padding: 10px;
  margin-top: 5px;
  margin-bottom: 5px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 80%;
  min-width: 32rem;
  background-color: #3CD3C1;
  border-radius: 8px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 2.0rem;
  margin-left: 30px;
}

.c7 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c8 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c3 {
  width: 10rem12rem8rem;
  height: 10rem12rem8rem;
  width: 10rem;
  height: 10rem;
  border-radius: 10rem;
  overflow: hidden;
}

.c4 {
  width: 100%;
  height: 100%;
}

@media screen and (min-width:40em) {
  .c2 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    gap: 3rem;
  }
}

@media screen and (min-width:52em) {
  .c6 {
    gap: 5rem;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    margin-left: 20px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    margin-left: 20px;
  }
}

@media screen and (min-width:40em) {
  .c3 {
    width: 12rem;
    height: 12rem;
  }
}

@media screen and (min-width:52em) {
  .c3 {
    width: 8rem;
    height: 8rem;
  }
}

<div
  class="c0"
  data-testid="listUsers"
>
  <a
    class="c1"
    cursor="pointer"
    width="80%"
  >
    <div
      class="c2"
    >
      <figure
        class="c3"
        height="10rem,12rem,8rem"
        width="10rem,12rem,8rem"
      >
        <img
          alt="alt"
          class="c4"
          src="https://randomuser.me/api/portraits/men/75.jpg"
        />
      </figure>
      <h2
        class="c5"
      >
        Jennie
      </h2>
    </div>
    <div
      class="c6"
    >
      <subtitle
        class="c7"
      >
        United States
      </subtitle>
      <body
        class="c8"
      >
        jennie.nichols@example.com
      </body>
    </div>
  </a>
  <a
    class="c1"
    cursor="pointer"
    width="80%"
  >
    <div
      class="c2"
    >
      <figure
        class="c3"
        height="10rem,12rem,8rem"
        width="10rem,12rem,8rem"
      >
        <img
          alt="alt"
          class="c4"
          src="https://randomuser.me/api/portraits/women/2.jpg"
        />
      </figure>
      <h2
        class="c5"
      >
        Ruby
      </h2>
    </div>
    <div
      class="c6"
    >
      <subtitle
        class="c7"
      >
        New Zealand
      </subtitle>
      <body
        class="c8"
      >
        ruby.jones@example.com
      </body>
    </div>
  </a>
</div>
`;
